src/test/ui/btreemap/btreemap_autotraits_without_lifetime.rs

```diff
@@ -0,0 +1,20 @@
+// check-fail
```

Comment:
I think most people don't bother to run the uitests locally for library patches, and the error message is the same as any other Sync error, so IMO its a bit better to do this as a `compile_fail` doctest on something private, and put `check-pass` in the normal tests.

That said this is unlikely to unexpectedly break in the future, so if you feel strongly that they should be a `uitest`, fine by me as-is.